node/src/chain/acceptor.rs

```diff
@@ -117,6 +119,47 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
         })
     }
 
+    /// Updates most_recent_block together with provisioners list.
+    ///
+    /// # Arguments
+    ///
+    /// * `blk` - Block that already exists in ledger
+    pub(crate) async fn update_most_recent_block(
+        &self,
+        blk: &Block,
+    ) -> anyhow::Result<()> {
+        let mut task = self.task.write().await;
+        let (_, public_key) = task.keys.clone();
+
+        let mut mrb = self.mrb.write().await;
+        let mut provisioners_list = self.provisioners_list.write().await;
+
+        // Ensure block that will be marked as blockchain tip does exist
+        let exists = self
+            .db
+            .read()
+            .await
+            .update(|t| t.get_block_exists(&blk.header.hash))?;
+
+        if !exists {
+            return Err(anyhow::anyhow!("could not find block"));
+        }
+
+        // Reset Consensus
+        task.abort_with_wait().await;
+
+        //  Update register.
+        self.db
+            .read()
+            .await
+            .update(|t| t.set_register(&blk.header))?;
```

Comment:
It's not related to this PR, but having a `lock.read()` that can perform `write` operations is quit weird

I know that rocksdb doesn't require a `mut`, but maybe we have to force it

Reply:
DB component implements interior mutability and rocksdb member is thread-safe. If we use `self.db.write()`  here it means that any concurrent read query will be blocked.

A few examples of concurrent read queries that we should not block:
- Block explorer requesting ledger data
- Responding to GetData/GetBlocks/GetTx kadcast messages on sync-up procedure.